controller/celebrant.controller.js

```diff
@@ -0,0 +1,253 @@
+const {pool} = require('../config/dbConnect');
+const { validationResult } = require('express-validator');
+const createCelebrantValidation = require('../utils/validator');
+
+
+//create a new celebrant
+const createCelebrant = async(req, res) => {
+
+     // Check for validation errors
+
+     const errors = validationResult(req);
+
+
+     if (!errors.isEmpty()) {
+         return res.status(400).json({ success: false, errors: errors.array() });
+     }
+
+    const {username,gender,phone_number,email,birthdate,channel,is_active} = req.body
+
+    try {
+         //check if celebrant exists
+         const celebrantExistsQuery = {text: "SELECT * FROM celebration.celebrants WHERE username = $1",
+         values: [username]
+        };
+
+        const { rows } = await pool.query(celebrantExistsQuery);
+
+        if (rows.length > 0) {
+            const celebrant = rows[0]
+
+        }
+
+        //insert new celebrant
+        const result = {
+            text: "INSERT INTO celebration.celebrants (username,gender,phone_number,email,birthdate,channel,is_active) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *",
+            values: [username,gender,phone_number,email,birthdate,channel,true]
+        }
+
+        const {rows: newCelebrant}= await pool.query(result);
+        
+         return res.status(200).json ({
+            success: true,
+            msg: "Celebrant created successfully",
+            data: newCelebrant[0]
+        });
+    } catch (error) {
+        console.error("Error creating new celebrant", error);
+        return res.status(500).json({
+          message: "internal server error",
+        });
+    }
+};
+
+
+//get a celebrant
+const getCelebrant = async (req, res) => {
+    const id = req.params.id;
+    try {
+        //fetch user
+         const getACelebrant = {text:'SELECT * FROM celebration.celebrants WHERE id = $1 AND is_deleted = false' ,
+         values: [id]
+        };
+
+        const { rows } = await pool.query(getACelebrant);
+
+        if (rows.length === 0) {
+            return res.status(404).json({ message: "Celebrant not found" });
+        }
+
+        res.status(200).json({
+            msg: "Successful!",
+            data : rows
+        });
+    } catch (error) {
+        console.error("Error fetching celebrant", error);
+        return res.status(500).json({
+          message: "internal server error",
+        });
+    }
+};
+
+//get all celebrants
+const getAllCelebrants = async (req, res) => {
+    try {
+        // Fetch all celebrants that are not marked as deleted
+        const getCelebrantsQuery = {
+            text: 'SELECT * FROM celebration.celebrants WHERE is_deleted = false'
+        };
+
+        const { rows } = await pool.query(getCelebrantsQuery);
+
+        if (rows.length === 0) {
+            return res.status(404).json({ message: "No celebrants found" });
```

Comment:
Why are you using different fields for the message in the error and success objects.

On line 97, you used `msg`, on line 93, you used `message`. Please unify this. Use `message` or `msg`, not both.

Reply:
done sir

thanks